### ISSUE TITLE
learning rate & weight decay

### ISSUE THREAD

**Issue by chenfh21** [PRBonn/phenobench-baselines#6]
Hi,
I am puzzled why the learning rate decay strategy during optimization in semantic segmentation benchmark methods is like this:
"During optimization, we employ Adam [9] and set the weight decay to 2 · 10−4. At the initial 16 epochs, we linearly increase the learning rate to 1 · 10−4 and subsequently apply a polynomial learning rate decay." refer to 'Phenobench Supplementary Material'
 Could you give me some thoughts about this?
![lr_image](https://github.com/user-attachments/assets/515eb2a9-ebc2-48cc-9f86-1696dabc1dc5)

**Comment by jbehley** [comment]
I think that the learning rate is increased from 0 to $1.0\cdot 10^{-4}$ for warm start and then decreased. I think the increase in the beginning is to avoid too noisy gradients in the beginning and then decrease the learning rate over time to converge to a minimum.

Does this answer you question? Maybe @JaWeyl can further provide some insights as he was the main developer of the baseline.

**Comment by chenfh21** [comment]
Thanks for your reply. I can understand this like yours. Maybe I need to ask @JaWeyl for further clarification. 








-----原始邮件-----
发件人:"Jens Behley" ***@***.***>
发送时间:2024-08-05 21:14:02 (星期一)
收件人: PRBonn/phenobench-baselines ***@***.***>
抄送: chenfh21 ***@***.***>, Author ***@***.***>
主题: Re: [PRBonn/phenobench-baselines] learning rate & weight decay (Issue #6)






I think that the learning rate is increased from 0 to $1.0\cdot 10^{-4}$ for warm start and then decreased. I think the increase in the beginning is to avoid too noisy gradients in the beginning and then decrease the learning rate over time to converge to a minimum.

Does this answer you question? Maybe @JaWeyl can further provide some insights as he was the main developer of the baseline.

**Comment by chenfh21** [comment]
> I think that the learning rate is increased from 0 to 1.0⋅10−4 for warm start and then decreased. I think the increase in the beginning is to avoid too noisy gradients in the beginning and then decrease the learning rate over time to converge to a minimum.
> 
> Does this answer you question? Maybe @JaWeyl can further provide some insights as he was the main developer of the baseline.

Thanks for your reply. I can understand this like yours. Maybe I need to ask @JaWeyl for further clarification.

**Comment by JaWeyl** [comment]
Hi @chenfh21,

The explanation of @jbehley is correct and provides an intuition about why we use the learning rate schedule.

**Comment by chenfh21** [comment]
> 嗨 ，
> 
> 的解释是正确的，并提供了关于我们为什么使用学习率时间表的直觉。
Thanks your reply. I have another question. I'm curious whether the baseline method provided in Phenobench's paper utilizes transfer learning/pre-trained weights.

**Comment by chenfh21** [comment]
> Hi @chenfh21,
> 
> The explanation of @jbehley is correct and provides an intuition about why we use the learning rate schedule.

Thanks for your reply. I have another question. I'm curious whether the baseline method provided in Phenobench's paper utilizes transfer learning/pre-trained weights.

**Comment by jbehley** [comment]
When we used pre-trained weights, we mention it in the supplement of the paper (see [PDF](https://www.ipb.uni-bonn.de/pdfs/weyler2024tpami.pdf) for the last version of the article with supplement included.), e.g., panoptic segmentation, leaf instance segmentation, uses pre-trained weights from ImageNet, but here we train from scratch for semantic segmentation.

**Comment by chenfh21** [comment]
> When we used pre-trained weights, we mention it in the supplement of the paper (see [PDF](https://www.ipb.uni-bonn.de/pdfs/weyler2024tpami.pdf) for the last version of the article with supplement included.), e.g., panoptic segmentation, leaf instance segmentation, uses pre-trained weights from ImageNet, but here we train from scratch for semantic segmentation.

Thanks for your reply very much.
